src/pytorch_lightning/loops/fit_loop.py

```diff
@@ -264,7 +264,9 @@ def advance(self) -> None:  # type: ignore[override]
         dataloader = self.trainer.train_dataloader
         assert self._data_fetcher is not None
         self._data_fetcher.setup(
-            dataloader, batch_to_device=partial(self.trainer._call_strategy_hook, "batch_to_device", dataloader_idx=0)
+            dataloader,
+            pre_batch_to_device=partial(self.trainer.lightning_module._on_before_batch_transfer, dataloader_idx=0),
+            batch_to_device=partial(self.trainer._call_strategy_hook, "batch_to_device", dataloader_idx=0),
```

Comment:
I don't think this should add a new argument, insted the `batch_to_device` here should be a local function that wraps both calls.

Reply:
done!